src/main/scala/Validator.scala

```diff
@@ -12,10 +12,23 @@ import scala.collection.mutable
 import scala.collection.mutable.{Map, Set}
 import scala.jdk.CollectionConverters.{IterableHasAsScala, MapHasAsScala}
 
-class Validator(var tableCsvFile: URI, sourceUri: String = "") {
+class Validator(var schemaUri: String, sourceUri: String = "") {
   val mapAvailableCharsets = Charset.availableCharsets().asScala
+  private def getAbsoluteSchemaUri(schemaPath: String): URI = {
+    val inputSchemaUri = new URI(schemaPath)
+    if (inputSchemaUri.getScheme == null) {
+      new URI(s"file://${new File(schemaPath).getAbsolutePath}")
+    } else {
+      inputSchemaUri
+    }
+  }
+
   def validate(): WarningsAndErrors = {
-    Schema.loadMetadataAndValidate(tableCsvFile) match {
+    // When CSV is fetched from web and the associated schema is in the header, this wont work.
+    // Change this (returning empty warnings and errors when schemaUri is blank) when that feature is implemented
+    if (schemaUri.isBlank) return WarningsAndErrors()
```

Comment:
Can we make `schemaUri: Option[String]` instead?

Reply:
Ya, changed it